### ISSUE TITLE
Device mismatch in PGD PyTorch

### ISSUE THREAD

**Issue by manishbcd** [Trusted-AI/adversarial-robustness-toolbox#1073]
**Describe the bug**
When running on the GPU, PGD (PyTorch) complains about operands being on different devices:

'# Apply mask
if mask is not None:
      grad = torch.where(mask == 0.0, torch.tensor(0.0), grad)

Looks like torch.tensor(0.0) should be mapped to self.estimator.device (since mask and grad are).

https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/1.6.1/art/attacks/evasion/projected_gradient_descent/projected_gradient_descent_pytorch.py

**Comment by beat-buesser** [comment]
Hi @manishbcd 

Thank you very much for reporting this issue! We'll fix it with 1.6.2.
